src/main/java/com/github/prominence/openweathermap/api/request/RequestSettings.java

```diff
@@ -22,66 +22,55 @@
 
 package com.github.prominence.openweathermap.api.request;
 
-import com.fasterxml.jackson.databind.JsonSerializer;
 import com.github.prominence.openweathermap.api.conf.TimeoutSettings;
-import com.github.prominence.openweathermap.api.core.net.HttpClient;
+import com.github.prominence.openweathermap.api.context.ApiConfiguration;
 import com.github.prominence.openweathermap.api.enums.Language;
 import com.github.prominence.openweathermap.api.enums.ResponseType;
 import com.github.prominence.openweathermap.api.enums.UnitSystem;
 
 import java.util.HashMap;
 import java.util.Map;
+import java.util.Optional;
 
 public class RequestSettings {
```

Comment:
It is a valid concern for the `.asJava()` terminator. It might still make sense for the other 3 options. To be honest, I would be tempted to leave it there as an optional thing (maybe with an appropriate comment). Alternatively, it would be possible to terminate the flow with the unit system in case of JSON/HTML/XML in one of the following ways:

1. `.asHTML(UnitSystem.METRIC);`
2. `.asHTML().unitSystem(UnitSystem.METRIC);` 

What do you think?

Reply:
I'm sorry, I've missed several comments from this PR.

What have you decided in that case? Does the problem is still actual in a new PR?

Reply:
No problem! This is not solved even in the new PR, I think option 1 would be cleaner overall and we will need a unit system either way in case of HTML, XML, JSON.

Reply:
Yes, the first variant will be more cleaner and easier to implement. Let's try it.